src/util/message.h

```diff
@@ -44,10 +45,47 @@ struct messaget
     template <typename... Args>
     static void println(FILE *f, VerbosityLevel lvl, Args &&...args)
     {
-      if(lvl == VerbosityLevel::Error)
-        fmt::print(f, "ERROR: ");
-      fmt::print(f, std::forward<Args>(args)...);
-      fmt::print(f, "\n");
+    	     
+      	/** 
+		* Newly added code to address issue #886.
+			* It prepends a well formatted timestamp, file path, function name, line number and log level to each LOG message. 
+			* It is transparent to the existing logging function, i.e., log_message.
+			* TODO: It can be implemented using a separate function and invoked here and/or elsewhere when required. 		
+		*/		
+		//========Start===============================================================================================
+		struct tm* time_struct; //A time structure consisting of 9 members, such as tm_sec, tm_min, tm_hour, tm_mday, tm_mon, tm_year, etc. 
+		char time_info[20]; //A variable to hold the date and time values.
+		const char *log_level;//[12]; //A variable to hold the log level.
+		
+		time_t current_time = time(0);
+		time_struct = localtime(&current_time);
+		strftime (time_info, sizeof(time_info), "%Y-%m-%d %H:%M:%S", time_struct); //"strftime" is to format the date/time structure. 
+	   
+		switch(lvl){//To prepend the respective log level to each log message.  
+			case VerbosityLevel::Error:
+				log_level = "[ERROR]:";
+				break;
+			case VerbosityLevel::Result:				
+				log_level = "[RESULT]:";
+				break;
+			case VerbosityLevel::Warning:				
+				log_level = "[WARNING]:";
+				break;
+			case VerbosityLevel::Progress:				
+				log_level = "[PROGRESS]:";
+				break;
+			case VerbosityLevel::Status:				
+				log_level = "[STATUS]:";
+				break;
+			default:
```

Comment:
Please, implement all the enum values in the switch, so we can remove this default.